spotipy/client.py

```diff
@@ -107,10 +107,16 @@ def _internal_call(self, method, url, payload, params):
         if not url.startswith('http'):
             url = self.prefix + url
         headers = self._auth_headers()
-        headers['Content-Type'] = 'application/json'
 
-        if payload:
-            args["data"] = json.dumps(payload)
+        if 'content_type' in args['params']:
+            headers['Content-Type'] = args['params']['content_type']
+            del args['params']['content_type']
+            if payload:
+                args["data"] = payload
+        else:
+            headers['Content-Type'] = 'application/json'
+            if payload:
+                args["data"] = json.dumps(payload)
```

Comment:
At first glance I thought we could write this in a better way, but then now I can't seem to express this any better either.